docs/index.asciidoc

```diff
@@ -160,6 +160,26 @@ fields:
 * `oq: bobo`
 * `ss: 12345`
 
+[id="plugins-{type}s-{plugin}-field_split_pattern"]
+===== `field_split_pattern`
+
+  * Value type is <<string,string>>
+  * There is no default value for this setting.
+
+A regex expression to use as field delimiter for parsing out key-value pairs.
```

Comment:
probably important note: the pattern expressions provided may NOT have:
 - captured groups (named or otherwise, as captures are used internally)
 - lookaheads or lookbehinds (as they themselves are used in lookaheads internally)
 - positional anchors (e.g., start-of-line `^`, end-of-line `$`, start-of-input `\A`, end-of-input `\Z`; as they conflict with how we scan input)

Reply:
+1

Reply:
Huh. Your example has a positive lookahead and it works? I didn’t know you could put a lookahead in a lookahead.

🌈💫#TheMoreYouKnow

Reply:
I added a spec using this patter: `"field_split_pattern" => ";\s*(?=key.+?:)|;$",` and it works.   A positive lookahead: `?=` and the end-of-line anchor: `;$` ... 

maybe we could rephrase that as more as a YMMV using these constructs? So we know capture groups are definitely not a good idea, lookaheads or lookbehinds and positional anchors YMMV. 
WDYT?

I am thinking we could also output the resulting regex as as debug line so users could take it and run some tests in regex testers until we roll out our own regex UI ... eventually ;)

Reply:
+1

The debug line will look a bit funky, because while composing regex from other regex makes the composition easier to read, each embedded subexpression includes the options that were present on the embedded pattern so there’s a lot of [noise](https://stackoverflow.com/a/28631019).

Reply:
yeah - but I think it could be useful nonetheless for debugging purposes.